dashboard/app/helpers/pd/workshop_survey_results_helper.rb

```diff
@@ -350,4 +513,53 @@ def get_summary_for_form(form_id, workshop)
 
     summary
   end
+
+  def should_only_show_current_user?
+    current_user &&
+      current_user.facilitator? &&
+      !(current_user.program_manager? ||
+        current_user.workshop_organizer? ||
+        current_user.workshop_admin?)
+  end
+
+  def reduce_summary(summary)
+    # Gnarly piece of code to reduce the summary from
+    # {
+    #   'Day 1' => {
+    #     general: {
+    #       q1: {...histogram...},
+    #       q2: {...histogram...}
+    #     },
+    #     facilitator: {
+    #       q3: {
+    #         'Facilitator 1' => {...histogram...},
+    #         'Facilitator 2' => {...histogram...}
+    #       }
+    #     }
+    #   },
+    #   'Day 2' => {
+    #     q4: {...histogram...},
+    #     q1: {...histogram...}
+    #     etc...
+    #   }
+    # }
+    # to
+    # {
+    #   q1: <merge of Day 1 and Day 2 histograms>
+    #   q2: q2 histogram
+    #   q3: {'Facilitator 1' => <merge of Day 1 and Day 2 q3 answers>, 'Facilitator 2' => ...}
+    #   q4: q4 histogram
+    # }
+
+    # First divide the questions by facilitator specific and facilitator non-specific.
+    facilitator_specific_questions = summary.select {|x| x.values.first.values.first.is_a? Hash}.map {|x| x.slice(*QUESTIONS_FOR_FACILITATOR_AVERAGES_QUESTION_LIST)}
+    general_questions = summary.reject {|x| x.values.first.values.first.is_a? Hash}.map {|x| x.slice(*QUESTIONS_FOR_FACILITATOR_AVERAGES_QUESTION_LIST)}
+
+    # Now reduce them all to one big hash. When we merge one histogram with the other,
+    # resolve conflicts by taking the sum of the two values
+    reduced_facilitator_questions = facilitator_specific_questions.reduce {|memo, obj| memo.merge(obj) {|_, o, n| o.merge(n) {|_, o1, n1| o1.merge(n1) {|_, o2, n2| o2 + n2}}}} || {}
```

Comment:
I think replacing o and n with old and new would make this longer but easier to understand - I had to read the documentation of merge and reduce several times through to figure out how this works.

Reply:
I think I can write this a little more leglibly